server/src/main/java/org/opensearch/gateway/GatewayAllocator.java

```diff
@@ -188,12 +199,27 @@ protected static void innerAllocatedUnassigned(
     public AllocateUnassignedDecision explainUnassignedShardAllocation(ShardRouting unassignedShard, RoutingAllocation routingAllocation) {
         assert unassignedShard.unassigned();
         assert routingAllocation.debugDecision();
-        if (unassignedShard.primary()) {
-            assert primaryShardAllocator != null;
-            return primaryShardAllocator.makeAllocationDecision(unassignedShard, routingAllocation, logger);
+        boolean batchMode = routingAllocation.nodes().getMinNodeVersion().onOrAfter(Version.CURRENT);
```

Comment:
batchMode should be checked from the setting introduced by @Gaurav614  in https://github.com/opensearch-project/OpenSearch/pull/8746 

`EXISTING_SHARDS_ALLOCATOR_BATCH_MODE_ENABLED`

Reply:
Using a batchMode flag added in #8746